java/code/src/com/suse/manager/errata/advisorymap/ErrataAdvisoryMapManager.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     * parses an advisory map csv file, saving the entries in the database
+     * <p>
+     * the advisory map csv file has the following structure:
+     * a) first row has headers
+     * b) each row represents an entry
+     * c) each entry row contains fields, coma (",") separated
+     * d) first field is the advisory (e.g. "SUSE-SLE-Module-Basesystem-15-SP6-2025-1733")
+     * e) second field is the announcementId (e.g. "SUSE-RU-2025:01733-1")
+     * f) third field (optional) is the advisoryUri
+     * (e.g. "https://www.suse.com/support/update/announcement/2025/suse-ru-202501733-1")
+     *
+     * @param inputStreamIn the errata advisory map input stream
+     */
+    public void readAdvisoryMapPopulateDatabase(InputStream inputStreamIn) throws IOException {
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(inputStreamIn))) {
+            String line;
+            boolean firstLine = true;
+            while ((line = br.readLine()) != null) {
+                if (firstLine) {
+                    //skip first line (headers) and clear database
+                    advisoryMapFactory.clearErrataAdvisoryMap();
+                }
+                else {
+                    String[] advisoryItem = line.split(ADVISORY_MAP_CSV_DELIMITER);
+
+                    String advisory = (advisoryItem.length > 0 ? advisoryItem[0] : "");
+                    String announcementId = (advisoryItem.length > 1 ? advisoryItem[1] : "");
+                    String advisoryUri = (advisoryItem.length > 2 ? advisoryItem[2] : "");
+
+                    ErrataAdvisoryMap advisoryMapEntry = new ErrataAdvisoryMap(advisory, announcementId, advisoryUri);
+                    advisoryMapFactory.save(advisoryMapEntry);
+                }
+                firstLine = false;
+            }
+        }
+    }
+
+    /**
+     * parses an advisory map csv file
+     *
+     * @param advisoryMapFile the errata advisory map csv file
+     */
+    public void readAdvisoryMapPopulateDatabase(File advisoryMapFile) throws IOException {
+        InputStream inputStream = new FileInputStream(advisoryMapFile);
+        readAdvisoryMapPopulateDatabase(inputStream);
+    }
+
+    private File createSafeTempFile() throws IOException {
+        FileAttribute<Set<PosixFilePermission>> attr =
+                PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------"));
+        return Files.createTempFile("advisoryMap", ".csv", attr).toFile();
+    }
+
+    /**
+     * Downloads the advisory map
+     *
+     * @return the temporary downloaded file with the advisory map
+     */
+    public File downloadErrataAdvisoryMap() throws IOException {
+        URL advisoryMapURL = new URL(ErrataAdvisoryMapManager.ADVISORY_MAP_CSV_SOURCE_URL);
+        File tempFile = createSafeTempFile();
+
+        // Start downloading
+        FileUtils.copyURLToFile(advisoryMapURL, tempFile, connectionTimeoutMs, readTimeoutMs);
```

Comment:
Maybe we should better use our HttpClientAdapter.
I have no idea if apache common handle all the special things like proxy and auth and proxyauth the same way as we do and if it has the passwords if needed.
Check SCCWebClient as it also download things from the web